rosbag2_compression/test/rosbag2_compression/test_sequential_compression_writer.cpp

```diff
@@ -384,6 +401,100 @@ TEST_P(SequentialCompressionWriterTest, writer_writes_with_compression_queue_siz
   EXPECT_EQ(fake_storage_size_, kNumMessagesToWrite);
 }
 
+
+TEST_P(SequentialCompressionWriterTest, writer_sets_nice_value)
+{
+  const std::string test_topic_name = "test_topic";
+  const std::string test_topic_type = "test_msgs/BasicTypes";
+  const uint64_t kCompressionQueueSize = GetParam();
+  const int wanted_nice_value = 10;
+
+  class TestCompressor : public rosbag2_compression::BaseCompressorInterface
+  {
+    int & detected_nice_value;
+
+public:
+    TestCompressor(int & detected_nice_value)
+    : detected_nice_value(detected_nice_value) {}
+    virtual std::string compress_uri(const std::string & uri)
+    {
+      return uri;
+    }
+
+    virtual void compress_serialized_bag_message(
+      const rosbag2_storage::SerializedBagMessage * bag_message,
+      rosbag2_storage::SerializedBagMessage * compressed_message)
+    {
+#ifdef _WIN32
```

Comment:
@jmachowinski The test passes on Linux only because there is a mistake with `ifdef _WIN32` statement.  
It shall be **ifndef _WIN32**

Reply:
The test is actually failing. I added the ifdef **after** running the test locally, which was stupid....

Reply:
@jmachowinski I said that 
> The selected approach for tests is not going to work.
Because currently test is trying to test against the fake `TestCompressor` instead of the real `SequentialCompressionWriter`. That doesn't make sense.

The unit tests shall call `SequentialCompressionWriter` where the changes are - directly or indirectly.

Reply:
@MichaelOrlov have a look at https://github.com/ros2/rosbag2/blob/64e29724cbea3058bfa4c1688867dc9f2f85bdbd/rosbag2_compression/test/rosbag2_compression/test_sequential_compression_writer.cpp#L111
This function is used inside the test, and initializes a SequentialCompressionWriter. I only use the factory, to provide a custom compressor, that is used within the compression thread. Therefore it is the perfect place to test for the correct priority.

Reply:
@jmachowinski Sorry, I overlooked it. The test was a bit messy. 
It does really make sense but needs a cleanup.
I've rewritten it. Please consider to merge or cherry-pick my changes from PR [Redesign test for the case when compression writer sets threads priority #1](https://github.com/cellumation/rosbag2/pull/1)